slu/slu/utils/config.py

```diff
@@ -172,6 +174,132 @@ def generate(self) -> Dict[str, Config]:
         return {config_dict[const.MODEL_NAME]: config}
 
 
+class YAMLPromptConfig(ConfigDataProviderInterface):
+    """
+    An instance of this class will load, validate and fetch the state <-> prompts mapping from prompts.yaml
+    The instance will be further passed to classifier plugins like xlmr.py.
+    """
+
+    def __init__(
+        self, config_path: Optional[str] = None, debug: Optional[bool] = True
+    ) -> None:
+        self.config_path: str = config_path or const.PROMPTS_CONFIG_PATH
+        self.null_prompt_token: str = "<pad>"
+        self.missing_nls_labels: set = set()
+        self.config_dict: dict[str] = {}
+        self.supported_languages: list[str] = []
+        self.debug: bool = debug
+
+    def get_config_dict(self, config_path: str) -> dict:
+        error_message = "Unable to read prompts.yaml file, ensure correct format."
+        with open(self.config_path, "r", encoding="utf8") as handle:
+            config_dict = yaml.safe_load(handle)
+            if not isinstance(config_dict, dict):
+                raise TypeError(error_message)
+            return config_dict
+
+    def get_supported_languages(self, config_dict: dict) -> List[str]:
+        supported_languages: list = list(config_dict.keys())
+        return supported_languages
+
+    def _get_config_path(self) -> str:
+        return self.config_path
+
+    def _valid_string(self, string: str) -> bool:
+        if isinstance(string, str):
+            if all(
+                [
+                    len(string) > 0,
+                    string != "nan",
+                    string != ".nan",
+                    string != "",
+                    string != " ",
+                    "Unnamed" not in string,
+                ]
+            ):
+                return True
+        return False
```

Comment:
You are using the same function in `setup-prompts.py` also. Please abstract it out as a util function and reuse it in both places.

Reply:
Created a new util function.